GCRCatalogs/dc2_object.py

```diff
@@ -229,14 +236,25 @@ def _subclass_init(self, **kwargs):
             self._columns = self._generate_columns(self._datasets)
 
         bands = [col[0] for col in self._columns if len(col) == 5 and col.endswith('_mag')]
-        self._quantity_modifiers = self._generate_modifiers(self.pixel_scale, bands)
+
+        # A slightly crude way of checking for version of schema to have modelfit mag
+        # A future improvement will be to explicitly store version information in the datasets
+        # and just rely on that versioning.
+        modelfit_mag_re = re.compile('.+_modelfit_mag')
+        has_modelfit_mag = [col for col in self._columns if modelfit_mag_re.match(col)]
```

Comment:
Line 243-244: regular expression seems overkill here; `endswith` is sufficient and more readable. Also, we can use `any` to allow short circuit (and also make code more direct in terms of readability)

```suggestion
        has_modelfit_mag = any(col.endswith('_modelfit_mag') for col in self._columns)
```